### PR TITLE
rgw: add roles_pool in RGWZoneParams dump/decode json

### PR DESCRIPTION
**Pull request by tianshan**
Signed-off-by: Tianshan Qu <tianshan@xsky.com>

### CODE REVIEW

File: src/rgw/rgw_json_enc.cc [hunk 1]
```diff
@@ -904,6 +904,7 @@ void RGWZoneParams::dump(Formatter *f) const
   encode_json("log_pool", log_pool, f);
   encode_json("intent_log_pool", intent_log_pool, f);
   encode_json("usage_log_pool", usage_log_pool, f);
+  encode_json("roles_pool", roles_pool, f);
```
Comment (amitkumar50) [inline src/rgw/rgw_json_enc.cc]: Would be good to write why roles_pool is added or create an tracker & link here for future reference.

Comment (tianshan) [reply src/rgw/rgw_json_enc.cc]: just provide a way to set roles_pool

**Comment by oritwas** [comment]
@tianshan , can you create a tracker issue and add it to the commit message.
This is needed for backport

**Comment by tianshan** [comment]
@oritwas  fixed.
issue is http://tracker.ceph.com/issues/22162

**Comment by cbodley** [comment]
http://pulpito.ceph.com/cbodley-2019-04-23_20:35:16-rgw-wip-cbodley-testing-distro-basic-smithi/

lots of valgrind issues from http://tracker.ceph.com/issues/39449

**Pull request merged**